autosklearn/estimators.py

```diff
@@ -821,3 +826,20 @@ def predict(self, X, batch_size=None, n_jobs=1):
 
     def _get_automl_class(self):
         return AutoMLRegressor
+
+
+def get_number_of_available_cores():
+    """
+    Returns the number of available logical cores on the system
+    First tries to use os.sched_getaffinity, if it is not available use os.cpu_count
+    :return: Number of cores, int
+    """
+
+    try:
+        n = len(os.sched_getaffinity(0))
+    except AttributeError:
+        n = os.cpu_count()
+
+    if n is None:
+        return 1
+    return n
```

Comment:
You might want to use `joblib.cpu_count()` that already does this and is aware of cgroups (e.g. docker CPU quotas).

Reply:
Thanks, good suggestion!
I replaced the custom function with the joblib function. The added unittest just make sure that they number of cores are used and it doesn't crash when using `n_jobs=-1`.